Modules/Module_Processing/Module_ASL/xASL_wrp_Quantify.m

```diff
@@ -187,70 +187,95 @@ function xASL_wrp_Quantify(x, PWI4D_Path, pathOutputCBF, M0Path, SliceGradientPa
-            elseif ~isempty(regexpi(sexOptions{iOption}, '^(female|f|woman|women)$'))
-                sexN(sex==iOption) = 2;
-            end
-        end
 
-        if sum(isnan(sexN))>0
-            warning('Unknown sex detected in participants.tsv, currently we can only infer hematocrit from male or female');
-            fprintf('%s\n', 'So in participants.tsv specify the words "male" and "female" only');
+% a2. Manage hematocrit usage parameter
+if isfield(x.modules.asl.bHct2BloodT1)
+    if x.modules.asl.bHct2BloodT1 == 2 && isempty(x.Q, 'Hematocrit')
+        warning('Parameter x.modules.asl.bHct2BloodT1 was set to 2: trying to infer hematocrit from age and sex, but hematocrit data were also found');
+        fprintf('%s\n', 'Consider setting x.modules.asl.bHct2BloodT1 to 1 to use the hematocrit data directly');
+    end
+    if x.modules.asl.bHct2BloodT1 == 2 && (isempty(IndexSetsAge) || isempty(IndexSetsSex))
+        warning('Parameter x.modules.asl.bHct2BloodT1 was set to 2: trying to infer hematocrit from age and sex, but age & sex data were incomplete');
+        fprintf('%s\n', 'Consider changing x.modules.asl.bHct2BloodT1 or ensure that age & sex data are present in participants.tsv');
+        x.modules.asl.bHct2BloodT1 = []; % Setting this to empty here, so it will be dealt with below
+    end
+
+if ~isfield(x.modules.asl.bHct2BloodT1) || isempty(x.modules.asl.bHct2BloodT1)
+    if isfield(x.Q, 'Hematocrit')
+        warning('Parameter x.modules.asl.bHct2BloodT1 was not set but Hematocrit data were found');
+        fprintf('%s\n', 'Setting x.modules.asl.bHct2BloodT1 to 1: converting hematocrit to blood T1 values');
+        fprintf('%s\n', 'Consider setting x.modules.asl.bHct2BloodT1 to avoid this warning');
+        x.modules.asl.bHct2BloodT1 = 1;
+    elseif ~isempty(IndexSetsAge) || ~isempty(IndexSetsSex)
+        warning('Parameter x.modules.asl.bHct2BloodT1 was not set but age & sex data were found');
+        fprintf('%s\n', 'Setting x.modules.asl.bHct2BloodT1 to option 2: trying to infer hematocrit from age & sex');
+        fprintf('%s\n', 'Consider setting x.modules.asl.bHct2BloodT1 to avoid this warning');
+        x.modules.asl.bHct2BloodT1 = 2;
+    else
+        fprintf('\n%s\n', 'No hematocrit data found, disabling x.modules.asl.bHct2BloodT1');
+        x.modules.asl.bHct2BloodT1 = 0;
+    end
+end
+
+
+% b. We model the expected hematocrit from age & sex
+if x.modules.asl.bHct2BloodT1 == 2
+    fprintf('%s\n', 'Trying to infer hematocrit from age & sex');
+    
+    age = x.S.SetsID(:, IndexSetsAge);
+    
+    % Convert sex correctly
+    sex = x.S.SetsID(:, IndexSetsSex);
+    sexOptions = x.S.SetsOptions{:, IndexSetsSex};
+    sexN = nan(length(sex), 1); % currently we can only infer hct from male/female
+    % So anything that is not detected, will remain NaNs
+    for iOption=1:length(sexOptions)
+        if ~isempty(regexpi(sexOptions{iOption}, '^(male|m|man|men)$'))
+            sexN(sex==iOption) = 1;
```

Comment:
I understand that previously, we used it for indexing. But here, we can use sex only internally and therefore we can use the 0/1 form and not 1/2. Or am I missing something?